resolwe_api/resolwe.py

```diff
@@ -407,52 +436,50 @@ def download(self, data_objects, field):
             raise ValueError("Only process results (output.* fields) can be downloaded")
 
         for o in data_objects:
-            o = str(o)
-            if re.match('^[0-9a-fA-F]{24}$', o) is None:
+            if re.match('^\d+$', str(o)) is None:
                 raise ValueError("Invalid object id {}".format(o))
 
-            if o not in self.cache['objects']:
-                self.cache['objects'][o] = Data(self.ta(o).get(), self)
+            if o not in self.cache['data']:
+                try:
+                    self.cache['data'][o] = Data(self.api.data(o).get(), self)
+                except HttpNotFoundError:
+                    raise ValueError("Data id {} does not exist".format(o))
 
-            if field not in self.cache['objects'][o].annotation:
-                raise ValueError("Download field {} does not exist".format(field))
+            if field not in self.cache['data'][o].annotation:
+                raise ValueError("Field {} does not exist for data object {}".format(field, o))
 
-            ann = self.cache['objects'][o].annotation[field]
+            ann = self.cache['data'][o].annotation[field]
             if ann['type'] != 'basic:file:':
-                raise ValueError("Only basic:file: field can be downloaded")
+                raise ValueError("Only basic:file: fields can be downloaded")
 
         for o in data_objects:
-            ann = self.cache['objects'][o].annotation[field]
+            ann = self.cache['data'][o].annotation[field]
             url = urlparse.urljoin(self.url, 'data/{}/{}'.format(o, ann['value']['file']))
-            yield requests.get(url, stream=True, auth=self.auth)
-
+            response = requests.get(url, stream=True, auth=self.auth)
+            if not response.ok:
+                response.raise_for_status()
+            else:
+                return response
 
 class ResAuth(requests.auth.AuthBase):
 
     """
     Attach HTTP Resolwe Authentication to Request object.
-
-    __init__ se pozene samo tkrat ko ustvaris ta objekt
-    __call__ se pa pozene vsakic ko ustvarjeni objekt poklices! Torej vedno razen prvic? tudi prvic?
-
     """
 
     def __init__(self, email=DEFAULT_EMAIL, password=DEFAULT_PASSWD, url=DEFAULT_URL):
-        payload = {
-            'username': email,
-            'password': password
-        }
+        payload = {'username': email, 'password': password}
 
         try:
             response = requests.post(urlparse.urljoin(url, '/rest-auth/login/'), data=payload)
-        except requests.exceptions.ConnectionError:
+        except ConnectionError:
             raise Exception('Server not accessible on {}'.format(url))
 
-        if response.status_code == 403:
+        if response.status_code in [400, 403]:
             raise Exception('Invalid credentials.')
```

Comment:
Use the following error message: Response HTTP status code {}. Invalid credentials?